core/io/resource_format_binary.cpp

```diff
@@ -1187,7 +1189,20 @@ Ref<Resource> ResourceFormatLoaderBinary::load(const String &p_path, const Strin
 	ERR_FAIL_COND_V_MSG(err != OK, Ref<Resource>(), "Cannot open file '" + p_path + "'.");
 
 	ResourceLoaderBinary loader;
-	loader.cache_mode = p_cache_mode;
+	switch (p_cache_mode) {
+		case CACHE_MODE_IGNORE:
+		case CACHE_MODE_REUSE:
+		case CACHE_MODE_REPLACE:
+			loader.cache_mode = p_cache_mode;
+			break;
+		case CACHE_MODE_IGNORE_DEEP:
+			loader.cache_mode = ResourceFormatLoader::CACHE_MODE_IGNORE;
+			break;
+		case CACHE_MODE_REPLACE_DEEP:
+			loader.cache_mode = ResourceFormatLoader::CACHE_MODE_REPLACE;
+			break;
+	}
+	loader.cache_mode_for_external = p_cache_mode;
```

Comment:
```suggestion
	switch (p_cache_mode) {
		case CACHE_MODE_IGNORE:
		case CACHE_MODE_REUSE:
		case CACHE_MODE_REPLACE:
			loader.cache_mode_for_external = CACHE_MODE_REUSE;
			loader.cache_mode = p_cache_mode;
			break;
		case CACHE_MODE_IGNORE_DEEP:
			loader.cache_mode_for_external = p_cache_mode;
			loader.cache_mode = ResourceFormatLoader::CACHE_MODE_IGNORE;
			break;
		case CACHE_MODE_REPLACE_DEEP:
			loader.cache_mode_for_external = p_cache_mode;
			loader.cache_mode = ResourceFormatLoader::CACHE_MODE_REPLACE;
			break;
	}
```
I assume it was supposed to be like this (https://github.com/godotengine/godot/pull/88664#issuecomment-1962430116).